connectors/es/sink.py

```diff
@@ -117,27 +141,24 @@ def __init__(
     ):
         self.client = client
         self.queue = queue
-        self.ops = defaultdict(int)
         self.chunk_size = chunk_size
         self.pipeline = pipeline
         self.chunk_mem_size = chunk_mem_size * 1024 * 1024
         self.bulk_tasks = ConcurrentTasks(max_concurrency=max_concurrency)
         self.max_retires = max_retries
         self.retry_interval = retry_interval
-        self.indexed_document_count = 0
-        self.indexed_document_volume = 0
-        self.deleted_document_count = 0
         self._logger = logger_ or logger
         self._canceled = False
         self._enable_bulk_operations_logging = enable_bulk_operations_logging
+        self.counters = Counters()
 
     def _bulk_op(self, doc, operation=OP_INDEX):
         doc_id = doc["_id"]
         index = doc["_index"]
 
         if operation == OP_INDEX:
             return [{operation: {"_index": index, "_id": doc_id}}, doc["doc"]]
-        if operation == OP_UPSERT:
```

Comment:
I'm not sure about getting rid of upsert - it's more resilient to weird errors or manual index manipulations. How will OP_UPDATE act if record's not present in the index?

Reply:
So, fun fact, `OP_UPSERT="update"`. See it in main:

https://github.com/elastic/connectors/blob/4047c2c6edf97ae8cfc9c846b3ec62e0c4f88157/connectors/es/sink.py#L57

This made me angry, so I renamed the variable to represent what the value actually was. Logically, nothing has changed.

Reply:
I can see why it was called "UPSERT" - cause it's indeed upsert, not an update - if document does not exist it's created - that would catch me off-guard if I was not familiar with how Elasticsearch handles it.

I don't oppose the rename, though :)

Reply:
the same is true with `OP_INDEX`. In fact, that's the better candidate for upsert, because `index` operations are expected to provide the full document, but `update` operations only require the piece of the document being updated. We don't make use of that, so there's really no need for us to use `update` operations at all. We just _count_ how many docs we're indexing that we already have IDs for as "updates", but we're actually using `OP_INDEX` for them. :/ It's a bit of a cluster.